pkg/job/runtime_v2/job/single/kube_single_job.go

```diff
@@ -126,21 +136,41 @@ func (sp *KubeSingleJob) customSingleJob(jobPod *v1.Pod, job *api.PFJob) error {
 		return fmt.Errorf("jobSpec or PFJob is nil")
 	}
 	// TODO: add more patch
+
+	if err := sp.setImageAndResource(&jobPod.Spec, job.Tasks[0]); err != nil {
```

Comment:
It might be better to split into `setImage` And `setResource`

Reply:
OK